File: sort/insertionSort/InsertionSort.java
import java.util.Random;

class InsertionSort{

	public static void insertionSort(int arr[]){

		int j, key;

		for(int i=1; i<arr.length; i++){
			key = arr[i];
			j = i-1;
			while( j >= 0 && key < arr[j] ){
				arr[j+1] = arr[j];
				j--;
			}
			arr[j+1] = key;
		}
	}

	public static void printArray(int arr[]){
		for(int x : arr)
			System.out.print(x + ", ");
	}

	public static void main(String args[]){

		int arr[] = new int[10];
		Random rand = new Random();

		for(int i=0; i<10; i++)
			arr[i] = rand.nextInt(100);

		System.out.print("\nOrignal Array : ");
		printArray(arr);

		insertionSort(arr);

		System.out.print("\n Sorted Array : ");
		printArray(arr);

		System.out.println("\n");
	}
}

File: sort/insertionSort/insertion_Sort.rs
fn insertion_sort(arr: &mut) {
    for i in 1..arr.len() {
        let mut j = i;
        while j > 0 && arr[j] < arr[j - 1] {
            arr.swap(j, j - 1);
            j = j - 1;
        }
    }
}

fn print_array(arr: &mut[i32]) {
    let size = arr.len();

    for i in 0..size {
        println!("{} ", arr[i]);
    }
    println!("\n");
}

fn main() {
    let mut array= [64, 34, 25, 12, 22, 11, 90];

    println!("Unsorted array: \n");
    print_array(&mut array);

    insertion_sort(&mut array);

    println!("Sorted array: \n");
    print_array(&mut array);
}